Backend: token service is not working

Token service is not working

```bash
[2025-12-07 06:22:31.705] [ERROR] [TokenService] ValidateRefreshToken failed: System.FormatException: The input string '"{\"id\":1,\"username\":\"Atomic\",\"role\":\"admin\",\"avatar\":\"/files/profile/ea551a378fe3b8510ab7fb2a86f12a0965c3c5b585b1ab4c05424faefb96f553.jpg\"}"' was not in a correct format.
   at System.Number.ThrowFormatException[TChar](ReadOnlySpan`1 value)
   at System.Int32.Parse(String s)
   at backend.Services.TokenService.ValidateRefreshToken(String refreshToken) in C:\Users\thoma\Documents\Event\backend\Service\TokenService.cs:line 115
```

Fix ASAP

--- a/backend/Service/TokenService.cs
+++ b/backend/Service/TokenService.cs
@@ -107,12 +107,32 @@ namespace backend.Services
         {
             try
             {
-                string? idString = await _cacheService.GetValueAsync($"refresh:{refreshToken}");
+                string? storedValue  = await _cacheService.GetValueAsync($"refresh:{refreshToken}");
 
-                if (string.IsNullOrEmpty(idString))
+                if (string.IsNullOrEmpty(storedValue ))
                     throw new UnauthorizedException("Invalid or expired refresh token.");
 
-                int userId = int.Parse(idString);
+                int userId;
+
+                if (int.TryParse(storedValue, out userId))
+                {
+                    // OK
+                }
+                else
+                {
+                    try
+                    {
+                        var user = JsonConvert.DeserializeObject<User>(storedValue)
+                            ?? throw new UnauthorizedException("Invalid refresh token payload.");
+
+                        userId = user.Id;
+                    }
+                    catch
+                    {
+                        throw new UnauthorizedException("Invalid refresh token payload.");
+                    }
+                }
+
 
                 var result = await _cacheService.DeleteKeyAsync($"refresh:{refreshToken}");
                 if (!result) throw new NotAvaliableException();

